app/src/main/java/com/example/erabook/AuthenticationViewModel.kt

```diff
@@ -1,22 +1,78 @@
 package com.example.erabook
 
+import android.content.Context
+import androidx.fragment.app.Fragment
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.bumptech.glide.Glide.init
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.firebase.FirebaseException
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.GoogleAuthProvider
 
+private const val RC_SIGN_IN = 9001
 class AuthenticationViewModel : ViewModel() {
     private val firebaseAuth = FirebaseAuth.getInstance()
-    val userLiveData: LiveData<FirebaseUser?> = MutableLiveData()
+    private val _userLiveData = MutableLiveData<FirebaseUser?>()
+    val userLiveData: LiveData<FirebaseUser?> = _userLiveData
+    private val _isTaskSuccessful = MutableLiveData<Boolean>()
+    val isTaskSuccessful:LiveData<Boolean?> = _isTaskSuccessful
+    private val _exception = MutableLiveData<String>()
+    val exception: LiveData<String> = _exception
+    private lateinit var mGoogleSignInClient: GoogleSignInClient
 
     init {
         firebaseAuth.addAuthStateListener { firebaseAuth ->
-            (userLiveData as MutableLiveData).value = firebaseAuth.currentUser
-
+            _userLiveData.value = firebaseAuth.currentUser
         }
     }
 
+    fun signInWithEmailAndPassword(email: String, password:String){
+        firebaseAuth.signInWithEmailAndPassword(email,password)
+            .addOnCompleteListener { task ->
+                if (task.isSuccessful){
+                    _isTaskSuccessful.value = true
+                }else{
+                    _isTaskSuccessful.value = false
+                    _exception.value = task.exception?.message
+                }
+            }
+    }
+
+    fun initializeGoogleSignInClient(context: Context){
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
```

Comment:
rename this please, don't use abbreviations for variables!
Name suggestion: googleSignInOptions